apps/choleskey/choleskey_stdpar.cpp

```diff
@@ -0,0 +1,126 @@
+// Cholesky Decomposition: stdpar
+#include "argparse/argparse.hpp"
+#include "commons.hpp"
+
+#include <algorithm>
+#include <execution>
+#include <experimental/mdspan>
+#include <iostream>
+#include <numeric>
+#include <vector>
+
+#include "matrixutil.hpp"
+
+using namespace std;
+
+// parameters
+struct args_params_t : public argparse::Args {
+  bool& results = kwarg("results", "print generated results (default: false)")
+                      .set_default(true);
+  std::uint64_t& nd =
+      kwarg("nd", "Number of inout matrix dimension").set_default(6);
+
+  bool& help = flag("h, help", "print help");
+  bool& time = kwarg("t, time", "print time").set_default(true);
+};
+
+///////////////////////////////////////////////////////////////////////////////
+
+struct solver {
+
+  using view_2d = std::extents<int, std::dynamic_extent, std::dynamic_extent>;
+
+  template <typename T>
+  std::vector<std::vector<T>> Cholesky_Decomposition(std::vector<T>& vec,
+                                                     int n) {
+    std::vector<std::vector<T>> lower(n, std::vector<T>(n, 0));
+
+    auto matrix_ms =
+        std::mdspan<T, view_2d, std::layout_right>(vec.data(), n, n);
+
+    auto multiplier_lambda = [=](auto a, auto b) {
+      return a * b;
+    };
+
+    // Decomposing a matrix into Lower Triangular
+    for (int i = 0; i < matrix_ms.extent(0); i++) {
+      for (int j = 0; j <= i; j++) {
+        T sum = 0;
+
+        if (j == i)  // summation for diagonals
+        {
+          sum = std::transform_reduce(
+              std::execution::par, lower[j].cbegin(), lower[j].cbegin() + j, 0,
+              std::plus{}, [=, i = i, j = j](int val) { return val * val; });
```

Comment:
It seems the i and j variables are not used in the lambda function. If true, we can remove it.

Reply:
okay. Thanks.